### ISSUE TITLE
Incorrectly flagging nested dynamic blocks as errors

### ISSUE THREAD

**Issue by centiqchris** [hashicorp/vscode-terraform#1594]
### Extension Version

v2.28.1

### VS Code Version

Version: 1.83.0
Commit: e7e037083ff4455cf320e344325dacb480062c3c
Date: 2023-10-03T16:11:49.843Z
Electron: 25.8.4
ElectronBuildId: 24154031
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Linux x64 6.2.0-34-generic

### Operating System

Ubuntu 22.04.3 LTS

### Terraform Version

Terraform v1.5.5 on linux_amd64

### Steps to Reproduce

1. Create a resource with one dynamic block nested inside another
2. Check errors highlighted by vscode

Trimmed Example
```
resource "azurerm_site_recovery_replicated_vm" "asr" {
  name                                      = "example"
  resource_group_name                       = "example"
  recovery_vault_name                       = "example"
  source_recovery_fabric_name               = "example"
  source_vm_id                              = "example"
  recovery_replication_policy_id            = "example"
  source_recovery_protection_container_name = "example"

  target_resource_group_id                = "example"
  target_recovery_fabric_id               = "example"
  target_recovery_protection_container_id = "example"
  target_network_id                       = "example"
  test_network_id                         = "example"

  dynamic "managed_disk" {
    for_each = local.alldisks
    content {
      disk_id                    = managed_disk.value.id
      staging_storage_account_id = var.vm_config.asr.staging_sa_id
      target_resource_group_id   = var.vm_config.asr.target_rg_id
      target_disk_type           = managed_disk.value.storage_account_type
      target_replica_disk_type   = managed_disk.value.storage_account_type

      dynamic "target_disk_encryption" {
        for_each = try(managed_disk.value.encryption_settings[0].disk_encryption_key[0].secret_url, null) != null ? { "this" = true } : {}
        content {
          disk_encryption_key {
            secret_url = data.azurerm_key_vault_secret.diskenc-asr[managed_disk.key].id
            vault_id   = try(var.vm_config.asr.key_vault_id, "fail")
          }

          key_encryption_key {
            key_url  = data.azurerm_key_vault_key.kek["this"].id
            vault_id = try(var.vm_config.asr.key_vault_id, "fail")
          }
        }
      }
    }
  }
}


```

### Expected Behavior

The syntax is correct and the code works as designed - no error should be reported.

### Actual Behavior

Creates the error `Unexpected block: Blocks of type "dynamic" are not expected here` and the inner keyword of dynamic is underlined red.

### Help Wanted

- [ ] I'm interested in contributing a fix myself

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by radeksimko** [comment]
@centiqchris Thanks for the report. I was able to reproduce this and I'm guessing this is the edge case related to blocks being treated as attributes and vice versa.

It's relatively uncommon but for mostly historical reasons, providers are allowed to mark certain attributes to be treated as blocks and the opposite.
https://github.com/hashicorp/terraform-provider-azurerm/blob/7574d9090cfffcb90c18ea0bac892e453ffe747e/internal/services/recoveryservices/site_recovery_replicated_vm_resource.go#L258

In other words, normally this would be two different things, where only one is valid:
```hcl
foo = [{}, {}] # attribute

# blocks
foo { }
foo { }
```
but the provider may choose to make _both_ valid.

It appears that we somehow prioritise attributes and fail to recognise duplicate blocks:
![Screenshot 2023-10-17 at 18 18 17](https://github.com/hashicorp/vscode-terraform/assets/287584/e3f0406c-d10f-4f0e-8805-24b99b83149e)

We'll have to check how this comes through in the provider schema.

**Comment by dbanck** [comment]
We just released a new version [2.29.0](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.29.0). This fixes the reported bug. The update should appear automatically in VS Code.

Thank you for the report @centiqchris 

--- 

In case you experience any different validation related bug, please do let us know through a new issue.
